### PR TITLE
 (PUP-9198) Add RHEL8 support in the dnf provider

### PR DESCRIPTION
**Pull request by mwhahaha**

### CODE REVIEW

**Comment by melissa** [comment]
It looks like the 5.5.x branch hasn't been merged up into 6.0.x or master yet. We should hold off merging this in until that merge up happens.

@mwhahaha this likely means you'll have some merge conflicts to deal with

**Comment by melissa** [comment]
For reference, the commit that went into 5.5.x is https://github.com/puppetlabs/puppet/pull/7140/commits/7ae809e1eaa826082c9f587f8741ea50f5246034

**Comment by joshcooper** [comment]
Thanks for your contribution @mwhahaha. For future reference, you'll only need to target the oldest branch, and we'll automatically merge the change up to more recent branches. We just merged the changes to 6.0.x in https://github.com/puppetlabs/puppet/commit/ddf87a7f1d69c94485156635671b391f484a6d42 (and will merge to master shortly) so I'm going to close this PR.

**Comment by joshcooper** [comment]
My bad, I thought I handled the switch from `defaultfor` to `notdefaultfor` in the merge up, but it doesn't look that way.

**Comment by mwhahaha** [comment]
@joshcooper ah my bad. I started with 6.0.x and goofed on the PR.  I'm used to a gerrit workflow where such things just work (and we work the other direction) :D This should be against 6.0.x now.

**Comment by joshcooper** [comment]
No worries @mwhahaha. It looks like there's now a conflict due to the 5.5.x PR being merged up to 6.0.x. Could you rebase and resolve conflicts?

File: lib/puppet/provider/package/dnf.rb [hunk 1]
```diff
@@ -30,7 +30,8 @@
 
   defaultfor :operatingsystem => :fedora
   notdefaultfor :operatingsystem => :fedora, :operatingsystemmajrelease => (19..21).to_a
-  defaultfor :osfamily => :redhat, :operatingsystemmajrelease => ["8"]
+  defaultfor :osfamily => :redhat
+  notdefaultfor :osfamily => :redhat, :operatingsystemmajrelease => ["6","7"]
```
Comment (joshcooper) [inline lib/puppet/provider/package/dnf.rb]: We still support [redhat 5](http://yum.puppet.com/puppet6/puppet6-release-el-5.noarch.rpm), so we should add it here.

File: lib/puppet/provider/package/yum.rb [hunk 2]
```diff
@@ -23,7 +23,7 @@
       end
   end
 
-  defaultfor :osfamily => :redhat
+  defaultfor :osfamily => :redhat, :operatingsystemmajrelease => ["6","7"]
```
Comment (joshcooper) [inline lib/puppet/provider/package/yum.rb]: Same comment.

**Comment by kris-bosland** [comment]
Thanks for your responsiveness on this PR @mwhahaha .  We would like to make sure this works for as many users as possible, so could you add RHEL 4 as well?

**Comment by mwhahaha** [comment]
@kris-bosland sure done

**Pull request merged**